src/DynamoCoreWpf/UI/Prompts/PresetPrompt.xaml.cs

```diff
@@ -24,6 +23,36 @@ public PresetPrompt()
 
         void OK_Click(object sender, RoutedEventArgs e)
         {
+            if (this.Owner != null)
+            {
+                var dataContext = this.Owner.DataContext as DynamoViewModel;
+                 //get the preset names from workspace
+                if (dataContext != null && dataContext.Model.CurrentWorkspace.Presets.Any())
```

Comment:
Why don't we store `dataContext.Model.CurrentWorkspace` in a temporary variable?